### PR TITLE
Tiny code cleanup: remove unnecessary comparisons

### PR DESCRIPTION
**Pull request by fzyzcjy**
The flags themselves are normal variables, so seems no need to check equality before assigning.

This PR is so small that I dare not ask for a test exemption or a review :P (Anyway I will continue working on flutter_smooth and those big PRs in a few days when I am not that busy)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
Hey @fzyzcjy, thanks for the contribution! Without a test we can't land this change, otherwise it could be reverted and we would be nine the wiser! 🙃 

Can you add a test?

**Comment by fzyzcjy** [comment]
@Piinks Sure, will do it later. Btw may I get some hints about how to add test for this feature? Not very familiar with registerBoolServiceExtension etc

**Comment by Piinks** [comment]
Hm, I am not really sure how to test this. Have you asked Hixie on Discord if it qualifies for a no test exemption?

**Comment by fzyzcjy** [comment]
@Piinks Sure, I will ask soon (when I find out a day to do all queued PRs and other tasks about Flutter - quite busy now indeed)

File: packages/flutter/lib/src/widgets/binding.dart
Comment (Hixie) [inline packages/flutter/lib/src/widgets/binding.dart:433]: this should be a body (`{ }`) rather than a return expression (`=>`) because otherwise it'll return bool instead of null. Might not matter, but better to be pedantic about this kind of thing since otherwise people might come to depend on it.

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

File: packages/flutter/lib/src/widgets/binding.dart
Comment (fzyzcjy) [reply packages/flutter/lib/src/widgets/binding.dart]: done

**Review by Piinks — APPROVED** [review]
LGTM! Thank you!

**Comment by Piinks** [comment]
Oh! Silly me, we need another reviewer. I will find one. :)

**Review by Renzo-Olivares — APPROVED** [review]
LGTM

**Comment by fzyzcjy** [comment]
You are welcome!

**Pull request merged**
